lms/djangoapps/course_goals/management/commands/goal_reminder_email.py

```diff
@@ -0,0 +1,100 @@
+    """
+    help = 'Send emails to users that are falling behind their course goals for the week'
+
+    def handle(self, *args, **options):
+        """
+        Docstring
+
+        Helpful notes for the function:
+            weekday() returns an int 0-6 with Monday being 0 and Sunday being 6
+        """
+        today = date.today()
+        sunday_date = today + timedelta(days=SUNDAY_WEEKDAY - today.weekday())
+        monday_date = today - timedelta(days=today.weekday())
+
+        # Monday is the start of when we consider user's activity towards counting towards their weekly
+        # goal. As such, we use Mondays to clear out the email reminders sent from the previous week.
+        if today.weekday() == MONDAY_WEEKDAY:
+            CourseGoal.objects.filter(email_reminder_sent=True).update(email_reminder_sent=False)
+            return
+
+        course_goals = CourseGoal.objects.filter(
+            days_per_week__gt=0, subscribed_to_reminders=True, email_reminder_sent=False,
+        )
+        all_goal_course_keys = course_goals.values_list('course_key', flat=True).distinct()
+        # Exclude all courses whose end dates are earlier than Sunday so we don't send an email about hitting
+        # a course goal when it may not even be possible.
+        courses_to_exclude = CourseOverview.objects.filter(
+            id__in=all_goal_course_keys, end__date__lte=sunday_date
+        ).values_list('id', flat=True)
+
+        course_goals = course_goals.exclude(course_key__in=courses_to_exclude).select_related('user').order_by('user')
+        with emulate_http_request(site=Site.objects.get_current()):  # emulate a request for waffle's benefit
+            for goal in course_goals:
+                self.handle_goal(goal, today, sunday_date, monday_date)
+
+    @staticmethod
+    def handle_goal(goal, today, sunday_date, monday_date):
+        if not COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled(goal.course_key):
+            return
+
+        enrollment = CourseEnrollment.get_enrollment(goal.user, goal.course_key, select_related=['course'])
+        # If you're not actively enrolled in the course or your enrollment was this week
+        if not enrollment or not enrollment.is_active or enrollment.created.date() >= monday_date:
+            return
+
+        audit_access_expiration_date = get_user_course_expiration_date(goal.user, enrollment.course_overview)
+        # If an audit user's access expires this week, exclude them from the email since they may not
+        # be able to hit their goal anyway
+        if audit_access_expiration_date and audit_access_expiration_date.date() <= sunday_date:
+            return
+
+        cert = get_certificate_for_user_id(goal.user, goal.course_key)
+        # If a user has a downloadable certificate, we will consider them as having completed
+        # the course and opt them out of receiving emails
+        if cert and cert.status == CertificateStatuses.downloadable:
```

Comment:
per my comment in slack i think Kaitlin may have said that we could check if a user has seen the course end page but i would check with her

Reply:
I'd like to delay implementing that (A) to double check with Kaitlin and (B) I think that's better as a follow-on split-off ticket - it feels like a separate chunk of work, because I don't think we currently track that at all.